### ISSUE TITLE
ipython crashes when home directory is on AWS EFS

### ISSUE THREAD

**Issue by erikcw** [ipython/ipython#9686]
I just transition a server's home directories to Amazon's new [EFS service](https://aws.amazon.com/efs/) (an NSF4 compatible fileshare), and now when attempting to launch ipython (version 4.2.0), I get the following traceback:

``` sh
$ mv .ipython .ipython.old
$ ipython
Traceback (most recent call last):
  File "/usr/local/bin/ipython", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python2.7/site-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python2.7/site-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/IPython/terminal/ipapp.py", line 305, in initialize
    self.init_shell()
  File "/usr/local/lib/python2.7/site-packages/IPython/terminal/ipapp.py", line 321, in init_shell
    ipython_dir=self.ipython_dir, user_ns=self.user_ns)
  File "/usr/local/lib/python2.7/site-packages/traitlets/config/configurable.py", line 405, in instance
    inst = cls(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 465, in __init__
    self.init_history()
  File "/usr/local/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 1586, in init_history
    self.history_manager = HistoryManager(shell=self, parent=self)
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 512, in __init__
    **traits)
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 204, in __init__
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
  File "<decorator-gen-14>", line 2, in init_db
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 97, in catch_corrupt_db
    self.init_db()
###
# Thousands of lines of this....
###
File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 90, in catch_corrupt_db
    return f(self, *a, **kw)
  File "/usr/local/lib/python2.7/site-packages/IPython/core/history.py", line 229, in init_db
    self.db = sqlite3.connect(self.hist_file, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/traitlets/traitlets.py", line 558, in __set__
    self.set(obj, value)
  File "/usr/local/lib/python2.7/site-packages/traitlets/traitlets.py", line 547, in set
    obj._notify_trait(self.name, old_value, new_value)
  File "/usr/local/lib/python2.7/site-packages/traitlets/traitlets.py", line 1093, in _notify_trait
    'type': 'change',
  File "/usr/local/lib/python2.7/site-packages/traitlets/traitlets.py", line 1115, in notify_change
    callables.append(_callback_wrapper(cb))
  File "/usr/local/lib/python2.7/site-packages/traitlets/traitlets.py", line 680, in _callback_wrapper
    return _CallbackWrapper(cb)
  File "/usr/local/lib/python2.7/site-packages/traitlets/traitlets.py", line 652, in __init__
    self.nargs = len(getargspec(cb)[0]) + offset
  File "/usr/local/lib/python2.7/site-packages/traitlets/utils/getargspec.py", line 70, in getargspec
    args, varargs, varkw = inspect.getargs(func.__code__)
  File "/usr/lib64/python2.7/inspect.py", line 800, in getargs
    return Arguments(args, varargs, varkw)
RuntimeError: maximum recursion depth exceeded

If you suspect this is an IPython bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@scipy.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    c.Application.verbose_crash=True
```

Moving the profile directory off of EFS fixes it:

``` sh
$ cp -R .ipython.old /tmp/
$ ipython --profile-dir /tmp/.ipython.old
Python 2.7.10 (default, Dec  8 2015, 18:25:23)
Type "copyright", "credits" or "license" for more information.

IPython 4.2.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

IPython profile: .ipython.old

In [1]:
```

**Comment by Carreau** [comment]
Hi there, 

Yes, we are aware of that, there is not much we can do: from `IPython --help-all`:

```
--HistoryManager.hist_file=<Unicode>
    Default: ''
    Path to file to use for SQLite history database.
    By default, IPython will put the history database in the IPython profile
    directory.  If you would rather share one history among profiles, you can
    set this value in each, so that they are consistent.
    Due to an issue with fcntl, SQLite is known to misbehave on some NFS mounts.
    If you see IPython hanging, try setting this to something on a local disk,
    e.g::
        ipython --HistoryManager.hist_file=/tmp/ipython_hist.sqlite
```

You can also use the specific `:memory:` value to keep the history in ram. (#9687 should add that to the documentation)

**Comment by Carreau** [comment]
Hum, strangely we have a `_corrupt_db_counter` that should do that on its own...

**Comment by Carreau** [comment]
> Hum, strangely we have a _corrupt_db_counter that should do that on its own...

Oh, actually that's only in master (#9418) , likely a duplicate of #9412, can you try with the current beta version ?

```
$ pip install ipython --pre --upgrade 
```

**Comment by Carreau** [comment]
Closing as this has likely be fixed by #9418
